Makefile

```diff
@@ -31,7 +30,33 @@ clean-lingering-resources:
 		&>/dev/null || true
 
 .PHONY: test
-test: ## Run tests using Helm chart-testing (ensure to run on dedicated test cluster)
+test: install-test-dependencies ## Run tests using Helm chart-testing (ensure to run on dedicated test cluster)
```

Comment:
chart-testing can not distinguish those. You either run all tests in the ci folder or you run all of them.

What we can split later though is to run them just for a single root chart, if I'm not mistaking.

E.g. some of those ci values would then move with the break out chart.

Also helm is in essence already integration testing as you always have a dependency on external system like k8s already.

Reply:
its not chart-testing that will use the other target. I think if we want to test a lot of the options, we can't stick to just integration testing. we will need unit testing like as proposed here: https://github.com/spiffe/helm-charts/pull/260/files#diff-a86124ab60e89d8dbec3b476b5a21077f58272f61f2007b0136c84e609738bb2

We can then do a lot of lighter weight tests to ensure config does get passed through properly, without having to deploy whole integration instance tests for every option.

Reply:
Whooaaa, for me that really feels like going a bridge too far with testing. I haven't seen any chart that is even testing on the level we already do. Most charts keep it even more to the basics not even testing with value overrides. I don't want to end up in the situation we have to maintain a whole bunch of self invented test approaches.

Are we trying to resolve how to develop and test Helm charts? Or are we trying to get a Helm chart for Spire.

IMHO such a unittesting thing needs an additional tool like chart-testing to make that usable without a lot of plumbing here.

Reply:
Its a much lighter weight way of testing so that the heavy weight things such as the integration tests don't need to be near as through. The go unit test stuff is basically one file with a few lines per test. Not particularly onerous.

Reply:
Revamped.